homeassistant/components/labs/websocket_api.py

```diff
@@ -108,3 +109,57 @@ async def websocket_update_preview_feature(
     hass.bus.async_fire(EVENT_LABS_UPDATED, event_data)
 
     connection.send_result(msg["id"])
+
+
+@callback
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "labs/subscribe",
+        vol.Required("domain"): str,
+        vol.Required("preview_feature"): str,
+    }
+)
+def websocket_subscribe_feature(
+    hass: HomeAssistant,
+    connection: websocket_api.ActiveConnection,
+    msg: dict[str, Any],
+) -> None:
+    """Subscribe to a specific lab preview feature updates."""
+    domain = msg["domain"]
+    preview_feature_key = msg["preview_feature"]
+    labs_data = hass.data[LABS_DATA]
+
+    preview_feature_id = f"{domain}.{preview_feature_key}"
+
+    if preview_feature_id not in labs_data.preview_features:
+        connection.send_error(
```

Comment:
When I look at this code, I think that `async_is_preview_feature_enabled ` should also have a check whether the lab feature exists at all and if it does not exist, we should throw an exception. This way we save people the trouble of dealing with typos and wondering why their code isn’t working. A similar issue that you’re trying to guard against on the frontend also shows up in Python code. Since this is a new API, it might be worth considering before more people start using it and we end up too scared to introduce changes.